server/service/service_campaigns.go

```diff
@@ -359,12 +372,56 @@ func (svc Service) updateStats(
 			return
 		}
 
+		tracker.lastStatsEntry = currentStats[0]
```

Comment:
Double checking: can `len(currentStats)` be `0` here?

Reply:
I don't think so, but I can add a check to be safe.